### PR TITLE
REST service case insensitive sub level search

### PR DESCRIPTION
**Pull request by Bournwog**
Problem: 
request `/rest/test/test2` search only in `Controllers/test/test2.php`
if path is like that `Controllers/Test/Test2.php` need to send this request `/rest/Test/Test2`

Fixed for case insensitive search, for all cases
`Controllers/test/test2.php`
`Controllers/Test/Test2.php`
`Controllers/test/Test2.php`
etc

### CODE REVIEW

**Comment by JoshuaLuckers** [comment]
@cla-bot check

File: core/src/Revolution/Rest/modRestService.php [hunk 1]
```diff
@@ -216,28 +216,42 @@ protected function getController()
...
-            if (strnatcasecmp($expectedFile, $controllerFile) == 0) {
-                require_once $basePath . $controller;
-
-                return $controllerClassPrefix . $controllerClassSeparator . $controllerClass;
-            }
-        }
+        
+        $iterateDirectories = $this->iterateDirectories($basePath . '/*' . $controllerClassFilePostfix, GLOB_NOSORT);
+    	if ($iterateDirectories) {           
+        	$contollersList = array();
```
Comment (JoshuaLuckers) [inline core/src/Revolution/Rest/modRestService.php]:
```suggestion
        	$contollersList = [];     
```

Comment (Ibochkarev) [reply core/src/Revolution/Rest/modRestService.php]: @Bournwog Up

**Review by Mark-H — APPROVED** [review]
I don't personally have an implementation of the rest service to properly test this with, but I don't see any issues with it from a code perspective.

**Comment by opengeek** [comment]
Like Mark, I do not have any implementations of this code to test. If we get another review approval, I'll assume it was tested appropriately and can be merged.

**Comment by Bruno17** [comment]
I could test it on 2.8.4 and it seems to solve case insensitive sub level search not working and works everything as expected.
Codewise, there isn't any difference on 3.x in this part, so I think it should work on both 2.8.x and 3.x

**Review by JoshuaLuckers — APPROVED** [review]
Based on @Bruno17's comment.

**Pull request merged**